app/javascript/components/App.vue

```diff
@@ -1,123 +1,160 @@
 <script setup>
-import { ref } from "vue";
-
+import { ref, computed } from "vue";
+import Spinner from "../components/Spinner.vue";
 const message = ref("");
+const data = ref(null);
 const error = ref(null);
 const loading = ref(false);
-const deps = ref([]);
-const items = ref([]);
-
-async function fetchData(url, valueName, depth) {
-    loading.value = true;
-    error.value = null;
-    items.value = [];
-    try {
-        items.value = await fetchRecursive(url, valueName, depth);
-    } catch (err) {
-        error.value = err.message;
-    } finally {
-        loading.value = false;
-    }
-}
-async function fetchRecursive(url, valueName, currentDepth) {
-    if (currentDepth) {
-        const response = await fetch(url, {
-            method: "POST",
-            headers: {
-                "Content-Type": "application/json;charset=utf-8",
-            },
-            body: JSON.stringify({ name: valueName }),
-        });
+const hasError = ref(null);
+const errorMessage = ref("");
 
-        if (!response.ok) {
-            throw new Error("Ошибка сети");
-        }
-        const data = await response.json();
-        const results = [];
+const inputClass = computed(() => {
+    return hasError.value ? "border-red-500 animate-shake" : "border-gray-300";
+});
 
-        if (data.dependencies) {
-            for (const [key, value] of Object.entries(data.dependencies)) {
-                if (!key.includes("@")) {
-                    const item = { name: key };
+const updateDataValue = computed(() => {
+    return message.value === "" ? (data.value = null) : "";
+});
 
-                    item.children = await fetchRecursive(
-                        url,
-                        key,
-                        currentDepth - 1
-                    );
+const handleSubmit = () => {
+    const englishPattern = /^[A-Za-z\s]+$/;
 
-                    results.push(item);
-                }
-            }
-        }
-        return results;
+    if (!message.value.trim()) {
+        hasError.value = true;
+        errorMessage.value = "Поле не должно быть пустым";
+    } else if (!englishPattern.test(message.value)) {
+        hasError.value = true;
+        errorMessage.value = "Пожалуйста, вводите только английские слова";
+    } else {
+        getDeps();
+        hasError.value = false;
+        errorMessage.value = "";
     }
-}
+    if (hasError.value) {
+        setTimeout(() => {
+            hasError.value = false;
+        }, 2000);
+    }
+};
+const getDeps = async () => {
+    loading.value = true;
+    data.value = null;
+
+    await fetch("http://localhost:3000/deps_collector/fetch_deps_all", {
```

Comment:
I think we can use path here with no host and port

Reply:
Fixed